src/fides/api/models/sql_models.py

```diff
@@ -514,6 +529,81 @@ def create(
         """Overrides base create to avoid unique check on `name` column"""
         return super().create(db=db, data=data, check_name=check_name)
 
+    @hybrid_property
+    def purpose(self) -> Optional[int]:
```

Comment:
i'm sort of curious how these get compiled into the generated SQL queries! if you have something handy i'd love to see, but no need to go out of your way.

Reply:
It's just a CASE statement added to the other SELECT attributes, it's not bad!  We can explore adding purpose too on the declaration at some point in the future. Cons are keeping data use/purpose in sync though.

```sql
SELECT
  ...
  CASE 
    WHEN (privacydeclaration.data_use = %(data_use_1)s) THEN %(param_1)s 
    WHEN (privacydeclaration.data_use = %(data_use_2)s) THEN %(param_2)s 
    WHEN (privacydeclaration.data_use = %(data_use_3)s) THEN %(param_3)s 
    WHEN (privacydeclaration.data_use = %(data_use_4)s) THEN %(param_4)s 
    WHEN (privacydeclaration.data_use = %(data_use_5)s) THEN %(param_5)s 
    WHEN (privacydeclaration.data_use = %(data_use_6)s) THEN %(param_6)s 
    WHEN (privacydeclaration.data_use = %(data_use_7)s) THEN %(param_7)s 
    WHEN (privacydeclaration.data_use = %(data_use_8)s) THEN %(param_8)s 
    WHEN (privacydeclaration.data_use = %(data_use_9)s) THEN %(param_9)s 
    WHEN (privacydeclaration.data_use = %(data_use_10)s) THEN %(param_10)s 
    WHEN (privacydeclaration.data_use = %(data_use_11)s) THEN %(param_11)s 
    WHEN (privacydeclaration.data_use = %(data_use_12)s) THEN %(param_12)s 
    WHEN (privacydeclaration.data_use = %(data_use_13)s) THEN %(param_13)s 
    WHEN (privacydeclaration.data_use = %(data_use_14)s) THEN %(param_14)s 
  END AS purpose,
  ....
```